homeassistant/components/enphase_envoy/config_flow.py

```diff
@@ -213,3 +213,69 @@ async def async_step_user(
             description_placeholders=description_placeholders,
             errors=errors,
         )
+
+    async def async_step_reconfigure(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Add reconfigure step to allow to manually reconfigure a config entry."""
+        errors: dict[str, str] = {}
+        description_placeholders: dict[str, str] = {}
+
+        entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        assert entry
+
+        host: Any = (user_input or entry.data).get(CONF_HOST)
+        username: Any = (user_input or entry.data).get(CONF_USERNAME)
+        password: Any = (user_input or entry.data).get(CONF_PASSWORD)
+
+        suggested_values = {
+            CONF_HOST: host,
+            CONF_USERNAME: username,
+            CONF_PASSWORD: password,
+        }
```

Comment:
why don't we just set `suggested_values` to `user_input or entry.data`? isn't that the same?

Reply:
Was thinking the token in entry.data may lead to unexpected effects but it doesn't. Changing it to what you propose leads to various linter type checking issues around the mapping proxy in entry.data that this construct avoids. If you have a suggest how to hanlde it I'll be happy to implement that.

Reply:
Hmm, I can think of a lot of ways on how to tackle this and none are perfect. 

What mapping proxy things you get?

Reply:
entry.data: MappingProxyType[str, Any]
user_input: dict[str, Any]
below works, 
suggested_values: dict[str, Any] | MappingProxyType[str, Any] = user_input or entry.data
but still need the indvidual elements for validate_input and _abort_if_unique_id_configured doesn't like the dict[str, Any] | MappingProxyType[str, Any] of suggested_values either.

Reply:
updated